src/status_im/contexts/wallet/wallet_connect/session_proposal/view.cljs

```diff
@@ -51,10 +51,49 @@
    {:source (quo.resources/get-network :optimism)}
    {:source (quo.resources/get-network :arbitrum)}])
 
+(defn- set-current-proposal-account
+  [acc]
+  (fn []
+    (rf/dispatch [:wallet-connect/set-current-proposal-account acc])
+    (rf/dispatch [:hide-bottom-sheet])))
+
+(defn- accounts-list
+  []
+  (let [accounts (rf/sub [:wallet/accounts-without-watched-accounts])
+        selected (rf/sub [:wallet-connect/current-proposal-account])]
+    [rn/view {:style style/account-switcher-list}
+     (for [account accounts]
+       ^{:key (-> account :address str)}
+       [quo/account-item
+        {:type          :default
+         :state         (if (and selected
+                                 (= (account :address)
+                                    (selected :address)))
+                          :selected
+                          :default)
+         :account-props account
+         :on-press      (set-current-proposal-account account)}])]))
```

Comment:
Should we make it a flatlist in case the user has excessively many accounts (or do we have a limit on nr of accounts in the wallet)?

Reply:
I don't think there is a limit on the number of accounts. If there are too many accounts, then we'll need a new design (with a proper view and not a bottom sheet). 

I'd suggest to let this be as it's unlikely that a beta user will have 100s of accounts. We can always change this later.

Reply:
agreed 👍